src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -589,7 +596,7 @@ SeparatedSyntaxList<SwitchExpressionArmSyntax> parseSwitchExpressionArms()
 
                     var savedState = _termState;
                     _termState |= TerminatorState.IsPatternInSwitchExpressionArm;
-                    var pattern = ParsePattern(Precedence.Coalescing, whenIsKeyword: true);
+                    var pattern = ParsePattern(Precedence.Coalescing, inSwitchArmPattern: true);
```

Comment:
one of hte two locations where 'true' is passed in.  Here we're parsing the arms for switch-expressions.